databases/week3/meal_sharing.sql

```diff
@@ -0,0 +1,128 @@
+CREATE DATABASE hyf_lesson3
+    DEFAULT CHARACTER SET = 'utf8mb4';
+    use hyf_lesson3;
+    CREATE TABLE `meal` (
+  `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `title` VARCHAR(255) NOT NULL,
+  `description` TEXT NOT NULL,
+  `location` VARCHAR(255) NULL,
+  `when` DATETIME,
+  `max_reservations` INT(10) NOT NULL,
+  `price` DECIMAL NOT NULL,
+  `created_date` DATE NULL
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `reservation`(
+  `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `number_of_guests` int(10) NOT NULL,
+  `created_date`DATE NULL,
+  `phone_number`VARCHAR(20) NULL,
+  `contact_name`VARCHAR(20) NOT NULL,
+  `email`VARCHAR(30) NOT NULL,
+  `meal_id`int(10) UNSIGNED NOT NULL,
+  CONSTRAINT `FK_meal` FOREIGN KEY (`meal_id`) REFERENCES `meal`(`id`) ON 
+  DELETE CASCADE ON UPDATE CASCADE
```

Comment:
If we delete all reservations for a given meal, the meal entry will also be deleted, is that what you want? (It is ok, just checking )

Reply:
yes,  if we use delete or update cascade the parent and child will act same right?

Reply:
The child will act the same, yes.